DOC: Added 'Modules Privacy Has Changed' section to overview.rst

aparimeet opened the pull request:
- [x] closes #22942 
- [x] tests passed

aparimeet committed:
Update overview.rst

Added 'Module Privacy Has Changed' section.

TomAugspurger commented on doc/source/overview.rst:
```diff
@@ -117,6 +117,46 @@ Institutional Partners
 
 The information about current institutional partners can be found on `pandas website page <https://pandas.pydata.org/about.html>`__.
 
+Modules Privacy Has Changed
```
Remove the "Has Changed"

aparimeet replied:
Sorry, didn't think about that. I'll get right on it.

TomAugspurger reviewed (commented):
High-level feedback: could you remove all references to "have been moved" or "has changed". The release note you based this off is meant for readers upgrading between versions of pandas. We want this to be a timeless document that describes things in the current version.

So instead of things like "... modules have been moved"  we really care about the parts like

```
The following modules in pandas are considered private:

* pandas.compat
* pandas.util
* pandas.core (and everything under it)
```

I don't think the table is necessary.

jschendel commented on doc/source/overview.rst:
```diff
@@ -117,6 +117,46 @@ Institutional Partners
+Modules Privacy Has Changed
+----------------------------
+
+Some formerly public python/c/c++/cython extension modules have been moved and/or renamed. These are all removed from the public API. Furthermore, the pandas.core, pandas.compat, and pandas.util top-level modules are now considered to be PRIVATE. If indicated, a deprecation warning will be issued if you reference theses modules. (GH12588)
```
Can you replace (GH12588) with (:issue:\`12588\`) here, and likewise with the other instances of this pattern elsewhere.  I don't think the GH pattern will actually link to the issues.

aparimeet replied:
Alright, I'll fix that.

aparimeet committed:
Added 'Modules Privacy' section to overview.rst 

Removed unnecessary text as was present in #23038.

aparimeet commented:
Let me know if this is correct.

TomAugspurger commented on doc/source/overview.rst:
```diff
@@ -117,6 +117,30 @@ Institutional Partners
+
+The following modules in pandas are considered PRIVATE:
+
+::
```
Does the double colon format this as a code block? If so, I don't think we need that. Sphinx should render the nested list fine.

TomAugspurger replied:
@aparimeet can you check how this renders without the double-colons? A regular nested list should be fine.

You may need to remove the line-breaks between top-level items, and put spaces before the `-`.

TomAugspurger commented on doc/source/overview.rst:
```diff
@@ -117,6 +117,30 @@ Institutional Partners
+Modules Privacy
+----------------
+
+The following modules in pandas are considered PRIVATE:
```
Two things

1. A bit of context would be helpful
2. I worry about this going out of date

So how about something like

    In addition to sub-packages and modules prefixed with `_`, the following sub-packages and modules are considered private and may change or functionality without warning.
    
    * `pandas.compat`
    * `pandas.core` and all submodules
    * `pandas.util` and all submodules

    While pandas developers have always considered everything in `pandas.core` "private", we deemed simply renaming it to `pandas._core` too disruptive.

    Pandas' public API consists of everything listed in :ref:`api`.

Alternatively, we can list out modules

TomAugspurger replied:
Agreed.

aparimeet committed:
Update overview.rst

aparimeet committed:
Update overview.rst

Edited into a nested list.

aparimeet committed:
Update overview.rst

TomAugspurger committed:
format, modules

TomAugspurger reviewed (approved):
Updated the modules.

I think we just need to list `pandas.core` and `pandas.compat`.

Thisn like `pandas.util.decorators` are deprecated, so there's no question whether they're in the public API.

jorisvandenbossche reviewed (commented):
I think the text around it suggested by Tom in https://github.com/pandas-dev/pandas/pull/23038#discussion_r223694209 would still be good to add

Do we have a better place for this than `overview.rst` ? It now feels a bit out of place between "Institutional partners" and "License"

TomAugspurger commented:
The neighboring sections are a bit strange, but "Package Overview" seems
like the right place.

Maybe we could expand the new section a bit with text describing the
modules that are public?

On Fri, Oct 26, 2018 at 3:38 AM Joris Van den Bossche <
notifications@github.com> wrote:

> *@jorisvandenbossche* commented on this pull request.
>
> I think the text around it suggested by Tom in #23038 (comment)
> <https://github.com/pandas-dev/pandas/pull/23038#discussion_r223694209>
> would still be good to add
>
> Do we have a better place for this than overview.rst ? It now feels a bit
> out of place between "Institutional partners" and "License"
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/23038#pullrequestreview-168722317>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHIlw25ytIpc1-gQGdbfbdnNYChqjxks5uosoPgaJpZM4XMZ1w>
> .
>

aparimeet commented:
Adding public modules to this would be a good idea.

aparimeet committed:
Added public modules

Changed 'Modules Privacy' to 'Private Modules' and added a 'Public Modules' section.

datapythonista commented:
I'm unsure about this. I see the value in documenting what is private or public, but:
- I agree that between partners and license doesn't seem to be the right place
- `api.rst` is already the list of what is public, and it's huge, we probably don't want to repeat this here, and I find quite confusing to have a public modules list with a subset

@TomAugspurger  What about moving this as a short `.. note::` section at the beginning of `api.rst` to warn about what is private instead?

TomAugspurger commented:
I notice now that we *do* have a note like that at the top of api.rst:
http://pandas-docs.github.io/pandas-docs-travis/api.html

On Tue, Nov 20, 2018 at 8:58 AM Marc Garcia <notifications@github.com>
wrote:

> I'm unsure about this. I see the value in documenting what is private or
> public, but:
>
>    - I agree that between partners and license doesn't seem to be the
>    right place
>    - api.rst is already the list of what is public, and it's huge, we
>    probably don't want to repeat this here, and I find quite confusing to have
>    a public modules list with a subset
>
> @TomAugspurger <https://github.com/TomAugspurger> What about moving this
> as a short .. note:: section at the beginning of api.rst to warn about
> what is private instead?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/23038#issuecomment-440302465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHIg-zcKqMpJr8j0Yin3KNY1DZTKeAks5uxBiigaJpZM4XMZ1w>
> .
>

datapythonista commented:
That's funny, didn't see that. Closing this then, feel free to reopen if you think something else is needed.

TomAugspurger commented:
Sorry for missing that @aparimeet. Let me know if you want help starting on
a new issue.

On Tue, Nov 20, 2018 at 9:06 AM Marc Garcia <notifications@github.com>
wrote:

> Closed #23038 <https://github.com/pandas-dev/pandas/pull/23038>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/23038#event-1977594802>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHIv8yHVuV8NbAV_ySP5gae0CLsLfMks5uxBp6gaJpZM4XMZ1w>
> .
>
